lib/src/rocket.rs

```diff
@@ -91,21 +93,40 @@ impl hyper::Handler for Rocket {
 #[cfg(not(feature = "tls"))]
 macro_rules! serve {
     ($rocket:expr, $addr:expr, |$server:ident, $proto:ident| $continue:expr) => ({
-        let ($proto, $server) = ("http://", hyper::Server::http($addr));
-        $continue
+        match $addr {
+            Address::Tcp(addr) => {
+                let ($proto, $server) = ("https://", hyper::Server::http(addr));
+                $continue
+            },
+            Address::Unix(path) => {
+                let $proto = "unix://";
+                #[cfg(unix)]
+                let $server = UnixSocketServer::new(path);
```

Comment:
Will this compile on non-Unix platforms? I think `$continue` uses `$server`, so you'll get a `use before declared` errors.